### PR TITLE
Add admin advisory banner to control unauthorized management of identity server

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Proposed changes in this pull request
- Added a new IdentityConnectorConfig type to handle and display an admin advisory banner in the admin login pages. This PR contains all the BE logic needed to store, change the given config. 
- Config is listed under the `Identity Provider` > `Resident` > `Login Attempt Security` >  `Admin Session Advisory Banner`
- There are two configs as listed below.
    - Enable Banner
    - Banner Description

### Related Issue
- https://github.com/wso2/product-is/issues/15521 

### Related PRs
- https://github.com/wso2/identity-apps/pull/3822
- https://github.com/wso2/carbon-identity-framework/pull/4480
- https://github.com/wso2/product-is/pull/15618

#### General

- [x] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [x] **Is the PR labeled correctly?**

#### Functionality

- [x] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [x] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [x] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [x] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [x] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [x] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [x] **Are all UI and API inputs run through forms or serializers?**
- [x] **Are all external inputs validated and sanitized appropriately?**
- [x] **Does all branching logic have a default case?**
- [x] **Does this solution handle outliers and edge cases gracefully?**
- [x] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                <groupId>com.github.spotbugs</groupId>
+                <artifactId>spotbugs-maven-plugin</artifactId>
+                <configuration>
+                    <threshold>High</threshold>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:143]: Missing newline at the end of the file

File: features/org.wso2.carbon.identity.admin.session.advisory.banner.control.server.feature/pom.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                            </bundles>
+                            <importFeatures>
+                            </importFeatures>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (hwupathum) [inline features/org.wso2.carbon.identity.admin.session.advisory.banner.control.server.feature/pom.xml:59]: Missing newline at the end of the file

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml [hunk 3]
```diff
@@ -0,0 +1,143 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2023, WSO2 Inc. (http://www.wso2.com).
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:3]: Should be WSO2 LLC

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.com).
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java:2]: Should be WSO2 LLC

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.com).
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java:2]: Should be WSO2 LLC

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import org.wso2.carbon.identity.admin.session.advisory.banner.control.constants.AdminSessionAdvisoryBannerControlConstants;
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class AdminSessionAdvisoryBannerControlHandler implements IdentityConnectorConfig {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java:27]: Missing class level comment

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import org.wso2.carbon.identity.admin.session.advisory.banner.control.handler.AdminSessionAdvisoryBannerControlHandler;
+import org.wso2.carbon.identity.governance.IdentityGovernanceService;
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+import org.wso2.carbon.user.core.service.RealmService;
+
+@Component(
+        name = "org.wso2.carbon.identity.admin.session.advisory.banner.control.internal" +
+                ".AdminSessionAdvisoryBannerControlComponent",
+        immediate = true)
+public class AdminSessionAdvisoryBannerControlComponent {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java:37]: Missing class level comment

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml (hunk 1 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:143]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: features/org.wso2.carbon.identity.admin.session.advisory.banner.control.server.feature/pom.xml (hunk 2 above)
Comment (mpmadhavig) [reply features/org.wso2.carbon.identity.admin.session.advisory.banner.control.server.feature/pom.xml:59]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml (hunk 3 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:3]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java (hunk 4 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java:2]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java (hunk 5 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java:2]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java (hunk 6 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/handler/AdminSessionAdvisoryBannerControlHandler.java:27]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java (hunk 7 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java:37]: resolved with 1bc0c4ef78c1f9502e402d0c5fd2494d51103f04

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>identity-governance</artifactId>
+        <groupId>org.wso2.carbon.identity.governance</groupId>
+        <version>1.8.21</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>org.wso2.carbon.identity.admin.session.advisory.banner.control</artifactId>
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:31]: have we finalized the artifact for this? org.wso2.carbon.identity.admin.session.advisory.banner.control is too long and too specific imo.
how about something broader like org.wso2.carbon.identity.server.admin.ui.control? @ashensw @madurangasiriwardena

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+            <artifactId>org.wso2.carbon.identity.event</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.datapublisher.authentication</groupId>
+            <artifactId>org.wso2.carbon.identity.data.publisher.application.authentication</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.event.handler.accountlock</groupId>
+            <artifactId>org.wso2.carbon.identity.handler.event.account.lock</artifactId>
+        </dependency>
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:109]: why do we need this for?

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+            <artifactId>org.wso2.carbon.identity.application.authentication.framework</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.event</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.datapublisher.authentication</groupId>
+            <artifactId>org.wso2.carbon.identity.data.publisher.application.authentication</artifactId>
+        </dependency>
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:105]: Do we use this dependency?

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.scr.ds-annotations</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.securevault</groupId>
+            <artifactId>org.wso2.securevault</artifactId>
+        </dependency>
+    </dependencies>
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml:119]: Please refactor these. I think we can get rid of some of these dependencies

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+            ".description";
+    public static final String ADMIN_SESSION_ADVISORY_BANNER_ENABLE_DEFAULT_VALUE = "false";
+    public static final String ADMIN_SESSION_ADVISORY_BANNER_DESCRIPTION_DEFAULT_VALUE = "Warning - unauthorized use " +
+            "of this tool is strictly prohibited. All activities performed using this tool are logged and monitored.";
+    public static final String HANDLER_NAME = "admin.session.advisory.banner.handler";
+    public static final String HANDLER_FRIENDLY_NAME = "Admin Session Advisory Banner";
+    public static final String HANDLER_CATEGORY = "Login Attempts Security";
+    public static final String HANDLER_SUB_CATEGORY = "DEFAULT";
+
+    public static final class AdminSessionAdvisoryBannerControlNameMapping {
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java:38]: let's add class comments

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml (hunk 9 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml]: Removed with 4372cf26aee9e86284ff687159f66a2a9b2aff23

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml (hunk 10 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml]: Removed with 4372cf26aee9e86284ff687159f66a2a9b2aff23

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml (hunk 11 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml]: Refactored with 4372cf26aee9e86284ff687159f66a2a9b2aff23

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java (hunk 12 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/constants/AdminSessionAdvisoryBannerControlConstants.java:41]: Added with 7f4f5a44062bdffc8cde70bbded0c85722829e78

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        } catch (RuntimeException e) {
+            log.error("Error while activating AdminSessionAdvisoryBannerControlHandler.", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Captcha Component is de-activated");
```
Comment (bhagyasakalanka) [inline components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java:66]: Update the message here

Comment (chamathns) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java:66]: if (log.isDebugEnabled())

this is redundant

Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/src/main/java/org/wso2/carbon/identity/admin/session/advisory/banner/control/internal/AdminSessionAdvisoryBannerControlComponent.java:66]: Resolved with 7f4f5a44062bdffc8cde70bbded0c85722829e78

File: components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml (hunk 8 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.admin.session.advisory.banner.control/pom.xml]: Resolved with 48d9b975f82157a67d31687da5d92d3454189663 
New component name: org.wso2.carbon.identity.admin.ui.banner.control

**Comment by mpmadhavig** [comment]
The implementation path has been changed and therefore the feature will be sent with the following effort.

1. https://github.com/wso2/carbon-identity-framework/pull/4512
2. https://github.com/wso2/carbon-kernel/pull/3560
3. https://github.com/wso2/identity-api-server/pull/441
4. https://github.com/wso2/identity-apps/pull/3872
5. https://github.com/wso2/product-is/pull/15700

**Pull request closed without merging**